ConnectionFactory.php

```diff
@@ -64,6 +68,24 @@ public function createConnection(array $params, Configuration $config = null, Ev
             $params     = $connection->getParams();
```

Comment:
I don't know why doing it in such convoluted way below. Isn't all we need to do instead is change this line to 
```suggestion
            $params     = $connectionOverrideOptions + $connection->getParams();
```
?

Reply:
The thought was to only allow specific overrides (`dbname`, `host`, `port`, `user` & `password`). https://github.com/doctrine/DoctrineBundle/pull/1290/commits/d1bbcb8f2ad708543269017af94194e4849bec2c streamlines the code by using `array_merge` - but in doing so will override / inject any connection param supplied in the `$connectionOverrideOptions` array. 

I'm good with reverting the aforementioned commit or keeping as is.